test/integration/protocols/liquidities.spec.ts

```diff
@@ -0,0 +1,289 @@
+      },
+    });
+  }
+
+  async function testTokenLiquidity({
+    ticker,
+    tokenAddress,
+    oracleSetup,
+    network,
+    slippage,
+  }: {
+    ticker: string;
+    tokenAddress: string;
+    oracleSetup: () => Promise<any>;
+    network: string;
+    slippage?: number;
+  }): Promise<void> {
+    let initialHubWETHBalance: BigNumber;
+    let reward: BigNumber;
+    let toProvide: BigNumber;
+    let token: IERC20;
+    describe(`WETH/${ticker}`, () => {
+      const WETH_ADDRESS = WETH_ADDRESS_BY_NETWORK[network];
+      given(async () => {
+        await snapshot.revert(snapshotId);
+        token = await ethers.getContractAt(IERC20_ABI, tokenAddress);
+        await oracleSetup();
+        await DCAHub.connect(cindy)['deposit(address,address,uint256,uint32,uint32,address,(address,uint8[])[])'](
+          WETH.address,
+          tokenAddress,
+          depositAmount,
+          AMOUNT_OF_SWAPS,
+          SwapInterval.ONE_MINUTE.seconds,
+          cindy.address,
+          []
+        );
+        const sortedTokens = tokenAddress < WETH_ADDRESS ? [tokenAddress, WETH_ADDRESS] : [WETH_ADDRESS, tokenAddress];
+        const wethIndex = tokenAddress < WETH_ADDRESS ? 1 : 0;
+        initialPerformedSwaps = await performedSwaps({ tokenAddress, wethAddress: WETH_ADDRESS });
+        initialHubWETHBalance = await WETH.balanceOf(DCAHub.address);
+        const { tokens } = await DCAHubCompanion.getNextSwapInfo([{ tokenA: sortedTokens[0], tokenB: sortedTokens[1] }]);
+        const weth = tokens[wethIndex];
+        const dexQuote = await zrx.quote({
+          chainId: 137,
+          sellToken: WETH_ADDRESS,
+          buyToken: tokenAddress,
+          sellAmount: weth.reward,
+          slippagePercentage: slippage ?? 0.005, // 0.5% as default
+          takerAddress: DCAHubCompanion.address,
+          skipValidation: true,
+        });
+        await DCAHubCompanion.connect(governor).defineDexSupport(dexQuote.to, true);
+        const swapTx = await DCAHubCompanion.swapWithDex(
+          dexQuote.to,
+          [sortedTokens[0], sortedTokens[1]],
+          [{ indexTokenA: 0, indexTokenB: 1 }],
+          [dexQuote.data],
+          false,
+          recipient.address,
+          constants.MAX_UINT_256
+        );
+
+        ({ reward, toProvide } = await getTransfers(swapTx));
+      });
+      then('swap is executed', async () => {
+        expect(await performedSwaps({ tokenAddress, wethAddress: WETH_ADDRESS })).to.equal(initialPerformedSwaps + 1);
+      });
+      // then('hub balance is correct', async () => {
+      //   const hubWETHBalance = await WETH.balanceOf(DCAHub.address);
+      //   const hubTokenBalance = await token.balanceOf(DCAHub.address);
+      //   expect(hubWETHBalance, 'Hub WETH balance is incorrect').to.equal(initialHubWETHBalance.sub(reward));
+      //   expect(hubTokenBalance, `Hub ${ticker} balance is incorrect`).to.equal(toProvide);
+      // });
```

Comment:
There is something going on with the MAI test here. We can check it out together later blz?

Reply:
Yeah, sure. Let's add it to our To Do list, so we don't forget